conf/aggregations.go

```diff
@@ -0,0 +1,112 @@
+package conf
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/alyu/configparser"
+)
+
+var DefaultAggregation = Aggregation{
+	Name:              "default",
+	Pattern:           nil,
+	XFilesFactor:      0.5,
+	AggregationMethod: []Method{Avg},
+}
+
+// Aggregations holds the aggregation definitions
+type Aggregations struct {
+	Data []Aggregation
+}
+
+type Aggregation struct {
+	Name              string
+	Pattern           *regexp.Regexp
+	XFilesFactor      float64
+	AggregationMethod []Method
+}
+
+// NewAggregations create instance of Aggregations
+func NewAggregations() Aggregations {
+	return Aggregations{
+		Data: make([]Aggregation, 0),
+	}
+}
+
+// ReadAggregations returns the defined aggregations from a storage-aggregations.conf file
+// and adds the default
+func ReadAggregations(file string) (Aggregations, error) {
+	config, err := configparser.Read(file)
+	if err != nil {
+		return Aggregations{}, err
+	}
+	sections, err := config.AllSections()
+	if err != nil {
+		return Aggregations{}, err
+	}
+
+	result := NewAggregations()
+
+	for _, s := range sections {
+		item := Aggregation{}
+		item.Name = strings.Trim(strings.SplitN(s.String(), "\n", 2)[0], " []")
+		if item.Name == "" || strings.HasPrefix(item.Name, "#") {
+			continue
+		}
+
+		item.Pattern, err = regexp.Compile(s.ValueOf("pattern"))
+		if err != nil {
+			return Aggregations{}, fmt.Errorf("[%s]: failed to parse pattern %q: %s", item.Name, s.ValueOf("pattern"), err.Error())
+		}
+
+		item.XFilesFactor, err = strconv.ParseFloat(s.ValueOf("xFilesFactor"), 64)
+		if err != nil {
+			return Aggregations{}, fmt.Errorf("[%s]: failed to parse xFilesFactor %q: %s", item.Name, s.ValueOf("xFilesFactor"), err.Error())
+		}
+
+		aggregationMethodStr := s.ValueOf("aggregationMethod")
+		methodStrs := strings.Split(aggregationMethodStr, ",")
+		for _, methodStr := range methodStrs {
+			switch methodStr {
+			case "average", "avg":
```

Comment:
good point. it's not documented in the config documentation either http://graphite.readthedocs.io/en/latest/config-carbon.html#storage-aggregation-conf ; so i guess metrictank will accept it because it is nice to its users.